### PR TITLE
Pass process list directly to ProcessListUpdateListener

### PR DESCRIPTION
**Pull request by dimitry-**
Also remove unused GetProcessList method

Test: builds

### CODE REVIEW

**Comment by dimitry-** [comment]
I think we still need mutex here but now to guard access to listener.

File: OrbitClientServices/ProcessManager.cpp [hunk 1]
```diff
@@ -57,8 +57,7 @@ class ProcessManagerImpl final : public ProcessManager {
   bool shutdown_initiated_;
 
   mutable absl::Mutex mutex_;
```
Comment (reichlfl) [inline OrbitClientServices/ProcessManager.cpp:59]: Nit: I'd prefer to have this named in a less generic way to indicate what it's supposed to be guarding.

**Review by reichlfl — APPROVED** [review]
LGTM, thanks. Maybe we should wait for https://github.com/google/orbit/pull/1526 to be merged before merging though.

**Review by antonrohr — APPROVED** [review]
LGTM. You do not need to wait for #1526.

File: OrbitClientServices/ProcessManager.cpp (hunk 1 above)
Comment (dimitry-) [reply OrbitClientServices/ProcessManager.cpp]: done.

File: OrbitClientServices/ProcessManager.cpp [hunk 2]
```diff
@@ -56,9 +56,8 @@ class ProcessManagerImpl final : public ProcessManager {
   absl::Mutex shutdown_mutex_;
   bool shutdown_initiated_;
 
-  mutable absl::Mutex mutex_;
-  std::vector<ProcessInfo> process_list_;
-  std::function<void(ProcessManager*)> process_list_update_listener_;
+  mutable absl::Mutex process_list_update_listener_mutex_;
```
Comment (dimitry-) [inline OrbitClientServices/ProcessManager.cpp:59]: and removed mutable.

**Pull request merged**